module_utils/ansible_nailgun_cement.py

```diff
@@ -340,15 +342,26 @@ def fields_equal(value1, value2):
     return value1 == value2
 
 
-def update_entity(old_entity, entity_dict, module, check_missing):
+def update_entity(old_entity, entity_dict, module, check_missing, check_type):
     try:
         volatile_entity = old_entity.read()
         result = volatile_entity
         fields = []
         for key, value in volatile_entity.get_values().items():
-            if key in entity_dict and not fields_equal(value, entity_dict[key]):
-                volatile_entity.__setattr__(key, entity_dict[key])
-                fields.append(key)
+            if key in entity_dict:
+                if check_type and not isinstance(value, type(entity_dict[key])):
+                    if isinstance(value, bool):
+                        entity_dict[key] = boolean(entity_dict[key])
+                    elif isinstance(value, list):
+                        entity_dict[key] = str(sorted(entity_dict[key].split(',')))
```

Comment:
Unfortunately, it seems this has to be a string, otherwise I get`Value is invalid: must be an array` which is a weird error because I'm sending an array..

```
2018-11-28T15:13:12 [I|app|8a4f3] Processing by Api::V2::SettingsController#update as JSON
2018-11-28T15:13:12 [I|app|8a4f3]   Parameters: {"setting"=>{"value"=>["bar", "foo"]}, "apiv"=>"v2", "id"=>"114"}
2018-11-28T15:13:12 [I|app|8a4f3] Authorized user admin(Admin User)
2018-11-28T15:13:12 [I|app|8a4f3] Current user set to admin (admin)
2018-11-28T15:13:12 [E|app|8a4f3] Unprocessable entity Setting::Provisioning (id: 114):
  Value is invalid: must be an array